awspricing/offers.py

```diff
@@ -54,8 +54,12 @@ def search_skus(self, **attributes):  # type: (**str) -> Set[str]
         :return: a set of matching SKUs
         """
         attributes = self._pythonify_attributes(attributes)
+        product_family = attributes.pop('productFamily', None)
```

Comment:
Perhaps there's a more common pattern of _optional_ attributes (test that they match only if present) that we should extract here? Seems like this may be a common ask.